airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -18,12 +18,13 @@
 """This module contains Amazon EKS operators."""
 import warnings
 from time import sleep
-from typing import Dict, Iterable, List, Optional
+from typing import Dict, Iterable, List, Optional, Union
 
 from airflow import AirflowException
 from airflow.models import BaseOperator
 from airflow.providers.amazon.aws.hooks.eks import ClusterStates, EksHook, FargateProfileStates
 from airflow.providers.cncf.kubernetes.operators.kubernetes_pod import KubernetesPodOperator
+from airflow.utils.types import NOTSET, ArgNotSet
```

Comment:
Would using these in providers force the minimum Airflow version required to be whatever version `airflow.utils.types.NOTSET/ArgNotSet` are released with?

Reply:
Yes

Reply:
Currently all providers are tested with importing  everything in 2.1 - so this will basically fail the provider test when rebased.

Reply:
dangit, min version strikes again!

Reply:
Ah yeah. That's price to pay for provider's flexibility, I am afraid.

Reply:
While they are in the same repo, you should really think of it like a Terraform provider which is in a separate repo and a separate source code, just depends on the core.

And especially for **utilities** not rely on the core package

Reply:
And the provider import tests helps with making sure it's not missed :D.

Reply:
OK i have replaced NOTSET with None

None works fine in this case because for KPO None is not allowed